rest/models/snapshot.go

```diff
@@ -378,3 +380,97 @@ type OrderBookQuote struct {
 	Price            float64            `json:"p,omitempty"`
 	ExchangeToShares map[string]float64 `json:"x,omitempty"`
 }
+
+type ListAssetSnapshotsParams struct {
+	TickerAnyOf *string `query:"ticker.any_of"`
+	Ticker      *string `query:"ticker"`
+
+	TickerLT  *string `query:"ticker.lt"`
+	TickerLTE *string `query:"ticker.lte"`
+	TickerGT  *string `query:"ticker.gt"`
+	TickerGTE *string `query:"ticker.gte"`
+
+	Type *string `query:"type"`
+}
+
+func (p ListAssetSnapshotsParams) WithTickerAnyOf(q string) *ListAssetSnapshotsParams {
+	p.TickerAnyOf = &q
+	return &p
+}
+
+func (p ListAssetSnapshotsParams) WithTicker(q string) *ListAssetSnapshotsParams {
+	p.Ticker = &q
+	return &p
+}
+
+func (p ListAssetSnapshotsParams) WithType(q string) *ListAssetSnapshotsParams {
+	p.Type = &q
+	return &p
+}
+
+func (p ListAssetSnapshotsParams) WithTickersByComparison(c Comparator, q string) *ListAssetSnapshotsParams {
+	switch c {
+	case LT:
+		p.TickerLT = &q
+	case LTE:
+		p.TickerLTE = &q
+	case GT:
+		p.TickerGT = &q
+	case GTE:
+		p.TickerGTE = &q
+	}
+	return &p
+}
+
+type ListAssetSnapshotsResponse struct {
+	BaseResponse
+	Results []SnapshotResponseModel `json:"results,omitempty"`
+}
+
+type SnapshotResponseModel struct {
+	Name         string            `json:"name,omitempty"`
+	MarketStatus string            `json:"market_status,omitempty"`
+	Ticker       string            `json:"ticker,omitempty"`
+	Type         string            `json:"type,omitempty"`
+	LastQuote    SnapshotLastQuote `json:"last_quote,omitempty"`
+	LastTrade    SnapshotLastTrade `json:"last_trade,omitempty"`
+	Session      Session           `json:"session,omitempty"`
+
+	BreakEvenPrice      float64         `json:"break_even_price,omitempty"`
+	Details             DetailsV2       `json:"details,omitempty"`
```

Comment:
We probably don't need the V2 on this, unless there's another type in this package with that name.